src/ontodia/widgets/treeNodes.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+
+import { FatClassModel } from '../diagram/elements';
+import { formatLocalizedLabel } from '../diagram/model';
+import { Node } from './node';
+
+export interface TreeNodesProps {
+    roots: ReadonlyArray<FatClassModel> | undefined;
+    expanded?: Boolean;
+    searchResult: Array<FatClassModel> | undefined;
+    searchString?: string | undefined;
+    lang?: Readonly<string> | undefined;
+    onClassSelected: (classId: string) => void;
+}
+
+const CLASS_NAME = 'ontodia-class-tree';
+
+export class TreeNodes extends React.PureComponent<TreeNodesProps, {}> {
+    public static defaultProps: Partial<TreeNodesProps> = {
+        expanded: true
+    };
+
+    constructor(props: TreeNodesProps) {
+        super(props);
+        this.filter = this.filter.bind(this);
+        this.compare = this.compare.bind(this);
+    }
+
+    filter(root: FatClassModel): Boolean {
+        const { searchResult } = this.props;
+        if (searchResult && searchResult.length > 0) {
+            return Boolean(searchResult.find(resultRoot => resultRoot === root));
+        } else {
+            return true;
+        }
+    }
+
+    compare(node1: FatClassModel, node2: FatClassModel) {
```

Comment:
This is an exact copy of the same comparison function in `classTree.tsx`?

Reply:
I got rid of sorting in classTree. She did not work.